cli-toolchain/graphql.md

```diff
@@ -1739,11 +1754,288 @@ mutation CreateProject {
+  @model(queries: null)
+  @key(name: "byPost", fields: ["postID", "editorID"])
+  @key(name: "byEditor", fields: ["editorID", "postID"]) {
+  id: ID!
+  postID: ID!
+  editorID: ID!
+  post: Post! @connection(fields: ["postID"])
+  editor: User! @connection(fields: ["editorID"])
+}
+
+type User @model {
+  id: ID!
+  username: String!
+  posts: [PostEditor] @connection(keyName: "byEditor", fields: ["id"])
+}
+```
+
+This case is a bidirectional many-to-many which is why two `@key` calls are needed on the PostEditor model.
+You can first create a Post and a User, and then add a connection between them with by creating a PostEditor object as follows:
+
+```graphql
+mutation CreateData {
+    p1: createPost(input: { id: "P1", title: "Post 1" }) {
+        id
+    }
+    p2: createPost(input: { id: "P2", title: "Post 2" }) {
+        id
+    }
+    u1: createUser(input: { id: "U1", username: "user1" }) {
+        id
+    }   
+    u2: createUser(input: { id: "U2", username: "user2" }) {
+        id
+    }
+}
+
+mutation CreateLinks {
+    p1u1: createPostEditor(input: { id: "P1U1", postID: "P1", editorID: "U1" }) {
+        id
+    }   
+    p1u2: createPostEditor(input: { id: "P1U2", postID: "P1", editorID: "U2" }) {
+        id
+    }
+    p2u1: createPostEditor(input: { id: "P2U1", postID: "P2", editorID: "U1" }) {
+        id
+    }
+}
+```
+
+Note that neither the User type nor the Post type have any identifiers of connected objects. The connection info is held entirely inside the PostEditor objects.
+
+You can query a given user with their posts:
+
+```graphql
+query GetUserWithPosts {
+    getUser(id: "U1") {
+        id
+        username
+        posts {
+            items {
+                post {
+                    title
+                }
+            }
+        }
+    }
+}
+```
+
+Also you can query a given post with the editors of that post and can list the posts of those editors, all in a single query:
+
+```graphql
+query GetPostWithEditorsWithPosts {
+    getPost(id: "P1") {
+        id
+        title
+        editors {
+            items {
+                editor {
+                    username
+                    posts {
+                        items {
+                            post {
+                                title
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+#### Alternative Definition
+
+The above definition is the recommended way to create relationships between model types in your API. This involves defining index structures using `@key` and connection resolvers using `@connection`. There is an older parameterization of @connection that creates indices and connection resolvers that is still functional for backwards compatibility reasons. It is recommended to use @key and the new @connection via the fields argument because it gives you more control and can do everything that the old @connection can do.
```

Comment:
I would remove "because it gives you more control and can do everything that the old @connection can do"